client/app/src/js/controllers/screenshots.js

```diff
@@ -17,10 +17,16 @@ function ScreenshotsController($stateParams, $state, $location, screenshots, Scr
 
   this.addScreenshot = function(url){
     Screenshot.addScreenshot(url)
-      .success(function(data){
-        console.log(data);
+      .then(function(data){
+        console.log('response on create', data);
+        // create an array if one doesnt exist, otherwise concat onto it
+        self.screenshots = Array.isArray(self.screenshots)
+          ? self.screenshots.concat([data.data])
+          : [data.data];
+
```

Comment:
So this is handling the case where `screenshots` resolves as `undefined`, yes? In that case, we could default to `[]` on line 9, then always use concat. (I think).